### PR TITLE
[ACR] Improve instructions for authenticating a client

### PR DESCRIPTION
**Pull request by annelo-msft**
Per [discussion](https://github.com/Azure/azure-sdk-for-net/pull/21493#discussion_r645773812) with @schaabs, we'd like to move away from documenting environment variables for AAD set-up and instead point users toward simpler options in the developer environment.  This update to the ACR readme makes a new proposal for the suggested changes.

### CODE REVIEW

File: sdk/containerregistry/Azure.Containers.ContainerRegistry/README.md [hunk 1]
```diff
@@ -23,49 +25,29 @@ dotnet add package Azure.Containers.ContainerRegistry --prerelease
...
 
 ```Powershell
-az acr create --name MyContainerRegistry --resource-group MyResourceGroup --location westus --sku Basic
+az acr create --name myregistry --resource-group myresourcegroup --location westus --sku Basic
 ```
 
 ### Authenticate the client
 
-The [Azure Identity library][identity] provides easy Azure Active Directory support for authentication.  You can [authenticate with the Azure CLI](https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/identity/Azure.Identity#authenticating-via-the-azure-cli) when developing and debugging locally.  Please see the [Azure Identity README][identity] for more approaches to authenticating with `DefaultAzureCredential`.
+For your application to connect to your registry, you'll need to create a `ContainerRegistryClient` that can authenticate with the registry service.  The [Azure Identity library][identity] provides easy Azure Active Directory support for authentication.
```
Comment (christothes) [inline sdk/containerregistry/Azure.Containers.ContainerRegistry/README.md:40]:
```suggestion
For your application to connect to your registry, you'll need to create a `ContainerRegistryClient` that can authenticate with the registry service.  The [Azure Identity library][identity] makes it easy to add Azure Active Directory support for authentication.  
```

File: sdk/containerregistry/Azure.Containers.ContainerRegistry/README.md [hunk 2]
```diff
@@ -92,6 +74,9 @@ We guarantee that all client instance methods are thread-safe and independent of
 
 ## Examples
 
+The following snippets show brief samples of common operations a user might use the ACR SDK library for.
```
Comment (christothes) [inline sdk/containerregistry/Azure.Containers.ContainerRegistry/README.md:77]:
```suggestion
The following snippets show brief samples of common developer scenarios using the ACR SDK library.  
```

**Review by christothes — APPROVED** [review]

**Review by schaabs — APPROVED** [review]
Awesome! I really like how you've organized the authentication concepts throughout the readme, looks great.

**Pull request merged**